src/main/kotlin/socket/achievement/PlayerUpdateListener.kt

```diff
@@ -0,0 +1,201 @@
+        blockCount: Int
+    ): PlayerContext {
+        reply(context.profile, PlayerUpdateData.DestroyableDestroyUpdateData(percentage, blockCount), PlayerUpdateReason.DESTROYABLE_DESTROY)
+        return super.onDestroyableDestroy(context, percentage, blockCount)
+    }
+
+    override suspend fun onCoreLeak(context: PlayerContext, percentage: Float, blockCount: Int): PlayerContext {
+        reply(context.profile, PlayerUpdateData.CoreLeakUpdateData(percentage, blockCount), PlayerUpdateReason.CORE_LEAK)
+        return super.onCoreLeak(context, percentage, blockCount)
+    }
+
+    override suspend fun onFlagPlace(context: PlayerContext, heldTime: Long): PlayerContext {
+        reply(context.profile, PlayerUpdateData.MonumentPlaceUpdateData(heldTime), PlayerUpdateReason.FLAG_PLACE)
+        return super.onFlagPlace(context, heldTime)
+    }
+
+    override suspend fun onFlagDrop(context: PlayerContext, heldTime: Long): PlayerContext {
+        reply(context.profile, PlayerUpdateData.MonumentDropUpdateData(heldTime), PlayerUpdateReason.FLAG_DROP)
+        return super.onFlagDrop(context, heldTime)
+    }
+
+    override suspend fun onFlagPickup(context: PlayerContext): PlayerContext {
+        reply(context.profile, PlayerUpdateData.NoArgs, PlayerUpdateReason.FLAG_PICKUP)
+        return super.onFlagPickup(context)
+    }
+
+    override suspend fun onFlagDefend(context: PlayerContext): PlayerContext {
+        reply(context.profile, PlayerUpdateData.NoArgs, PlayerUpdateReason.FLAG_DEFEND)
+        return super.onFlagDefend(context)
+    }
+
+    override suspend fun onWoolPlace(context: PlayerContext, heldTime: Long): PlayerContext {
+        reply(context.profile, PlayerUpdateData.MonumentPlaceUpdateData(heldTime), PlayerUpdateReason.WOOL_PLACE)
+        return super.onWoolPlace(context, heldTime)
+    }
+
+    override suspend fun onWoolPickup(context: PlayerContext): PlayerContext {
+        reply(context.profile, PlayerUpdateData.NoArgs, PlayerUpdateReason.WOOL_PICKUP)
+        return super.onWoolPickup(context)
+    }
+
+    override suspend fun onWoolDrop(context: PlayerContext, heldTime: Long): PlayerContext {
+        reply(context.profile, PlayerUpdateData.MonumentDropUpdateData(heldTime), PlayerUpdateReason.WOOL_DROP)
+        return super.onWoolDrop(context, heldTime)
+    }
+
+    override suspend fun onWoolDefend(context: PlayerContext): PlayerContext {
+        reply(context.profile, PlayerUpdateData.NoArgs, PlayerUpdateReason.WOOL_DEFEND)
+        return super.onWoolDefend(context)
+    }
+
+    override suspend fun onControlPointCapture(context: PlayerContext, contributors: Int): PlayerContext {
+        reply(context.profile, PlayerUpdateData.ControlPointCaptureUpdateData(contributors), PlayerUpdateReason.CONTROL_POINT_CAPTURE)
+        return super.onControlPointCapture(context, contributors)
+    }
+
+    private suspend fun reply(player: Player, data: PlayerUpdateData, reason: PlayerUpdateReason) {
```

Comment:
for consistency, I think this method should be moved to the PlayerContext class (where other MC-bound player-oriented sockets are sent from, such as message sending or XP gains), as an `update` method for example. also, if you do this, the `server` param can be removed from the constructor of this listener (even if this method stays here, I believe that constructor param could be replaced by accessing the server from `PlayerContext#match`)